apps/spruce/src/pages/image/tabs/EventLogTab/useEvents.ts

```diff
@@ -0,0 +1,62 @@
+import { useState, useEffect } from "react";
+import { useQuery } from "@apollo/client";
+import { useToastContext } from "context/toast";
+import {
+  ImageEventsQuery,
+  ImageEventsQueryVariables,
+} from "gql/generated/types";
+import { IMAGE_EVENTS } from "gql/queries";
+import { IMAGE_EVENT_LIMIT } from "../../ImageEventLog";
+
+export const useEvents = (
+  imageId: string,
+  page: number = 0,
+  limit: number = IMAGE_EVENT_LIMIT,
+) => {
+  const dispatchToast = useToastContext();
+
+  const { data, fetchMore, loading } = useQuery<
```

Comment:
Is it possible to follow the `useDistroEvents` function more closely? I'm a bit confused why it looks so different. For example, could we make a call to the `useEvents` hook and keep track of `prevCount` there?